lnrpc/invoicesrpc/invoices_server.go

```diff
@@ -47,16 +73,48 @@ func fileExists(name string) bool {
 	return true
 }
 
-// New returns a new instance of the invoicesrpc Invoices sub-server. We also
+// New returns a new instance of the Invoices sub-server. We also
 // return the set of permissions for the macaroons that we may create within
 // this method. If the macaroons we need aren't found in the filepath, then
 // we'll create them on start up. If we're unable to locate, or create the
 // macaroons we need, then we'll return with an error.
 func New(cfg *Config) (*Server, lnrpc.MacaroonPerms, error) {
-	// We don't create any new macaroons for this subserver, instead reuse
-	// existing onchain/offchain permissions.
+	// If the path of the invoices macaroon wasn't specified, then we'll
+	// assume that it's found at the default network directory.
+	macFilePath := filepath.Join(
+		cfg.NetworkDir, DefaultInvoicesMacFilename,
+	)
+
+	// Now that we know the full path of the invoices macaroon, we can
+	// check to see if we need to create it or not.
+	if !fileExists(macFilePath) && cfg.MacService != nil {
```

Comment:
Gave it a try, but it gets many parameters. I think it may be better to, in a separate pr, have subservers expose some macaroon info and bake the macaroon somewhere else.

Reply:
sounds good, thanks!